js/bitmex.js

```diff
@@ -287,144 +425,177 @@ module.exports = class bitmex extends Exchange {
-        //        "indicativeSettlePrice": 143.807,
+        //        "indicativeSettlePrice": "19136.27",
         //        "optionUnderlyingPrice": null,
         //        "settledPriceAdjustmentRate": null,
         //        "settledPrice": null,
-        //        "timestamp": "2022-01-14T17:49:55.000Z"
-        //    }
+        //        "timestamp": "2022-10-16T14:52:20.000Z"
+        //      },
+        //     ]
         //
         const result = [];
         for (let i = 0; i < response.length; i++) {
             const market = response[i];
             const id = this.safeString (market, 'symbol');
-            const baseId = this.safeString (market, 'underlying');
-            const quoteId = this.safeString (market, 'quoteCurrency');
-            const settleId = this.safeString (market, 'settlCurrency', '');
-            const base = this.safeCurrencyCode (baseId);
-            const quote = this.safeCurrencyCode (quoteId);
-            const settle = this.safeCurrencyCode (settleId);
-            const basequote = baseId + quoteId;
-            const swap = (id === basequote);
+            const typ = this.safeString (market, 'typ'); // type definitions at: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            const underlying = this.safeString (market, 'underlying');
+            const quoteCurrency = this.safeString (market, 'quoteCurrency');
+            const settlCurrency = this.safeString (market, 'settlCurrency', '');
+            const base = this.safeCurrencyCode (underlying);
+            const quote = this.safeCurrencyCode (quoteCurrency);
+            const settle = this.safeCurrencyCode (settlCurrency);
+            const baseId = this.currencyIdFromAssetName (underlying);
+            const quoteId = this.currencyIdFromAssetName (quoteCurrency);
+            const settleId = this.currencyIdFromAssetName (settlCurrency);
             // 'positionCurrency' may be empty ("", as Bitmex currently returns for ETHUSD)
             // so let's take the settlCurrency first and then adjust if needed
             let type = undefined;
+            let spot = false;
             let future = false;
+            let swap = false;
             let prediction = false;
             let index = false;
-            let symbol = base + '/' + quote + ':' + settle;
+            let symbol = undefined;
             const expiryDatetime = this.safeString (market, 'expiry');
             const expiry = this.parse8601 (expiryDatetime);
-            const inverse = this.safeValue (market, 'isInverse');
             const status = this.safeString (market, 'state');
-            let active = status !== 'Unlisted';
-            if (swap) {
+            const active = status !== 'Unlisted';
+            let contract = false;
+            // types defined here: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            if (typ === 'IFXXXP') {
+                type = 'spot';
```

Comment:
are we lacking spot support inside `createorder` ? 
![image](https://user-images.githubusercontent.com/43336371/211792714-7364dc50-c2d2-47e7-a3b7-7a039b96170d.png)

Reply:
i think you refer to `reduceOnly` functionality, which is not needed for spot. you can just "sell" on stop.
did i miss smth?

Reply:
@ttodua no my bad, I was blind and didn't notice the `reduceOnly` if 😅